### PR TITLE
Transfers: fix checksum validation in FTS. Closes #5670 

### LINKED ISSUE #5670: Logic error in FTS3 transfertool checksum validation strategy
Motivation
----------
The function `checksum_validation_strategy` in `rucio/transfertool/fts3.py` determines which checksumming strategy to apply to a transfer. This function has some errors, for instance comparing the variables `source_supported_checksums` and `dest_supported_checksums` with `['none']`, instead of the correct value `[]`.

Another error is when the source supports checksumming, but the destination does not, which implies there are no common checksum algorithms. In this case case the checksum validation strategy is still selected as `destination`, and not `source`, despite the fact that the destination doesn't have any supported algorithms. This causes checksumming to be skipped on the source, even when it should have been performed.

Relevant source for context:

```python3
def checksum_validation_strategy(src_attributes, dst_attributes, logger):
    source_supported_checksums = get_rse_supported_checksums_from_attributes(src_attributes)
    dest_supported_checksums = get_rse_supported_checksums_from_attributes(dst_attributes)
    common_checksum_names = set(source_supported_checksums).intersection(dest_supported_checksums)

    ...

    if len(common_checksum_names) == 0:
        logger(logging.INFO, 'No common checksum method. Verifying destination only.')
        verify_checksum = 'destination'

    ...
```

Modification
------------
Determine the correct expected behaviour and rewrite `checksum_validation_strategy`.

### PR DESCRIPTION
**Pull request by rcarpa**
This PR build on top of the initial work by @akselaase in #5674. During the review I discovered multiple additional bugs, which are fixed in the additional commits.

### CODE REVIEW

File: lib/rucio/core/rse.py [hunk 1]
```diff
@@ -749,29 +741,12 @@ def parse_checksum_support_attribute(checksum_attribute):
     if not checksum_attribute:
         return GLOBALLY_SUPPORTED_CHECKSUMS
     else:
-        supported_checksum_list = checksum_attribute[0].split(',')
+        supported_checksum_list = [c for c in checksum_attribute.split(',') if c]
```
Comment (mlassnig) [inline lib/rucio/core/rse.py]: maybe add `.strip()` ?

Comment (rcarpa) [reply lib/rucio/core/rse.py]: done

**Comment by bari12** [comment]
Rebase please

**Review by bari12 — APPROVED** [review]

**Comment by bari12** [comment]
Thanks @rcarpa and @akselaase 👍

**Pull request merged**